src/plugins/navigation_embeddable/public/editor/navigation_embeddable_editor_tools.tsx

```diff
@@ -7,16 +7,12 @@
  */
 
 import { memoize } from 'lodash';
-import { NavigationEmbeddableLink, NavigationEmbeddableLinkList } from '../embeddable/types';
+import { NavigationEmbeddableLink } from '../../common/content_management';
 
-const getOrderedLinkList = (links: NavigationEmbeddableLinkList): NavigationEmbeddableLink[] => {
-  return Object.keys(links)
-    .map((linkId) => {
-      return links[linkId];
-    })
-    .sort((linkA, linkB) => {
-      return linkA.order - linkB.order;
-    });
+const getOrderedLinkList = (links: NavigationEmbeddableLink[]): NavigationEmbeddableLink[] => {
+  return [...links].sort((linkA, linkB) => {
+    return linkA.order - linkB.order;
+  });
```

Comment:
Just a note that this will be cleaned up as part of my design PR, which moves all the caching to the Dashboard CM service 🎉